Documentation/install/upgrade.rst

```diff
@@ -211,8 +211,11 @@ configuration options for each minor version.
 
   .. group-tab:: Helm
 
-    Keeping an existing `ConfigMap` with ``helm upgrade`` is currently not
-    supported.
+    .. parsed-literal::
+
+      helm upgrade |CHART_RELEASE| \\
+        --namespace kube-system cilium \\
+        --set global.configMap="$(kubectl get -n kube-system cm cilium-config -o yaml)"
```

Comment:
This is interesting. We shouldn't be using a global variable but and relying on the `kube-system` namespace is not 100% ideal either but it looks like a great quick fix.

As with other solutions, everything we are doing seems to be going against some Helm design principles. A user would probably expected that if another `--set` modifies a value in the ConfigMap, then that is respected as well. We are clearly hitting the limits of Helm as well.